fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/RecordValidationTest.java

```diff
@@ -0,0 +1,238 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.apple.foundationdb.record.provider.foundationdb;
+
+import com.apple.foundationdb.record.TestRecords1Proto;
+import com.apple.foundationdb.tuple.Tuple;
+import com.apple.test.BooleanSource;
+import com.google.common.base.Strings;
+import com.google.protobuf.Message;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import javax.annotation.Nonnull;
+import java.util.EnumSet;
+import java.util.List;
+import java.util.stream.Stream;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+public class RecordValidationTest extends FDBRecordStoreTestBase {
+    @Nonnull
+    public static Stream<Integer> formatVersions() {
+        return Stream.of(
+                FDBRecordStore.RECORD_COUNT_KEY_ADDED_FORMAT_VERSION, // 3
+                FDBRecordStore.SAVE_VERSION_WITH_RECORD_FORMAT_VERSION, // 6
+                FDBRecordStore.MAX_SUPPORTED_FORMAT_VERSION);
+    }
+
+    public static Stream<Arguments> splitAndVersion() {
+        return Stream.of(true, false)
+                .flatMap(
+                        split -> formatVersions()
+                                .map(version -> Arguments.of(split, version)));
+    }
+
+    @ParameterizedTest(name = "testValidateRecordsNoIssue [splitLongRecords = {0}], formatVersion = {1}")
+    @MethodSource("splitAndVersion")
+    public void testValidateRecordsNoIssue(boolean splitLongRecords, int formatVersion) throws Exception {
+        final RecordMetaDataHook hook = getRecordMetaDataHook(splitLongRecords);
+        List<FDBStoredRecord<Message>> result = saveRecords(splitLongRecords, formatVersion, hook);
+        // Validate by primary key
+        try (FDBRecordContext context = openContext()) {
+            final FDBRecordStore store = openSimpleRecordStore(context, hook, formatVersion);
+            validateRecord(store, result.get(0), null);
+            validateRecord(store, result.get(1), null);
+            context.commit();
+        }
+    }
+
+    @ParameterizedTest(name = "testValidateRecordsMissingRecord [splitLongRecords = {0}], formatVersion = {1}")
+    @MethodSource("splitAndVersion")
+    public void testValidateRecordsMissingRecord(boolean splitLongRecords, int formatVersion) throws Exception {
```

Comment:
You should probably test this both with and without leaving a version around.

Reply:
I think this is already covered by `missingVersion` and `missingSplit` tests. Do you see something missing?